csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/CsvValidatorUi.scala

```diff
@@ -543,24 +543,20 @@ object CsvValidatorUi extends SimpleSwingApplication {
         model.asInstanceOf[DefaultTableModel].addRow(rowData.asInstanceOf[Array[AnyRef]])
       }
 
-      def removeSelectedRow() : Unit = {
-        model.asInstanceOf[DefaultTableModel].removeRow(peer.getSelectedRow)
+      def removeSelectedRows() : Unit = peer.getSelectedRows.zipWithIndex.foreach { case (row, index) =>
+        model.asInstanceOf[DefaultTableModel]removeRow(row - index) // when you delete a row, the index of next row to delete shifts down by 1
       }
 
       def pathSubstitutions : List[(String, String)] = {
         for(rowIdx <- (0 to model.getRowCount - 1)) yield (model.getValueAt(rowIdx, 0).asInstanceOf[String], model.getValueAt(rowIdx, 1).asInstanceOf[String])
       }.toList
     }
 
-    private val popupMenu = new PopupMenu
-    private val miRemove = new MenuItem("Remove Path Substitution")
-    miRemove.reactions += onClick(tblPathSubstitutions.removeSelectedRow())
-    popupMenu.contents += miRemove
-    tblPathSubstitutions.popupMenu(popupMenu)
-
     private val spTblPathSubstitutions = new ScrollPane(tblPathSubstitutions)
+    private val btnRemovePathSubstitution = new Button("Remove Path Substitution...")
```

Comment:
Please remove the `...` from the text. The ellipsis in UI is a cue to user that there will be more input needed before the action is performed. In case of Removal, the action is performed as soon as the button is clicked so it should not have the ellipsis